src/sentry/demo/tasks.py

```diff
@@ -18,19 +24,49 @@ def delete_users_orgs(**kwargs):
     # delete everything older than a day
     cutoff_time = timezone.now() - timedelta(days=1)
 
-    # first mark orgs for deletion
     # note this only runs in demo mode (not SaaS) so the underlying tables here are small
     org_list = Organization.objects.filter(
-        date_added__lte=cutoff_time, flags=F("flags").bitor(Organization.flags["demo_mode"])
+        demoorganization__isnull=False,
+        demoorganization__date_assigned__lte=cutoff_time,
+        demoorganization__status=DemoOrgStatus.ACTIVE,
+        status__in=(
+            OrganizationStatus.ACTIVE,
+            OrganizationStatus.PENDING_DELETION,
+        ),
     )
+
+    # first mark orgs for deletion
     org_list.update(status=OrganizationStatus.PENDING_DELETION)
 
     # next delete the users
-    User.objects.filter(
-        date_joined__lte=cutoff_time, flags=F("flags").bitor(User.flags["demo_mode"])
-    ).delete()
+    User.objects.filter(demouser__isnull=False, demouser__date_assigned__lte=cutoff_time).delete()
 
     # now finally delete the orgs
     for org in org_list:
         # apply async so if so we continue if one org aborts
         delete_organization.apply_async(kwargs={"object_id": org.id})
+
+
+ORG_BUFFER_SIZE = 3
+
+
+@instrumented_task(
+    name="sentry.demo.tasks.build_up_org_buffer",
+)
+def build_up_org_buffer():
+    if not settings.DEMO_MODE:
+        return
```

Comment:
Not sure if we even need this check, we'll only ever run this task if the demo app is installed. Actually wondering if we need `settings.DEMO_MODE` at all,  but probably worth keeping for now just in case.

Reply:
@wedamija We probably _don't_ need the check but I am purposely writing the demo code so that one mistake doesn't accidentally enable things it shouldn't. Just an extra layer of precaution knowing how much demo mode would mess up SaaS lol.